stdlib/list.ml

```diff
@@ -77,6 +77,21 @@ let rec flatten = function
 
 let concat = flatten
 
+let intersperse sep =
```

Comment:
An implementation that takes advantage of `[@tail_mod_cons]` would look cleaner in my opinion. Just like we now have for `List.map` .

Reply:
Indeed! I've fixed this for both `intersperse` and `intercalate`. Thanks for pointing this out.